src/fmu/dataio/providers/objectdata/_base.py

```diff
@@ -175,83 +73,108 @@ class ObjectDataProvider(Provider):
     # result properties; the most important is metadata which IS the 'data' part in
     # the resulting metadata. But other variables needed later are also given
     # as instance properties in addition (for simplicity in other classes/functions)
-    metadata: dict = field(default_factory=dict)
+    _metadata: dict = field(default_factory=dict)
     name: str = field(default="")
-    classname: str = field(default="")
     efolder: str = field(default="")
     extension: str = field(default="")
     fmt: str = field(default="")
     time0: datetime | None = field(default=None)
     time1: datetime | None = field(default=None)
```

Comment:
These should eventually all become properties defined by each object's child provider. Most will likely return an enum value.